src/js/core.js

```diff
@@ -553,6 +554,27 @@ function MediumEditor(elements, options) {
         if (action === 'image') {
             return this.options.ownerDocument.execCommand('insertImage', false, this.options.contentWindow.getSelection());
         }
+        /* Issue: https://github.com/yabwe/medium-editor/issues/595
+         * If the action is to justify the text */
+        if (justifyAction.exec(action)) {
+            var reg = /<br\/*>/gm,
+                html = Selection.getSelectionHtml.call(this);
+            /* The problem is caused by the <br> that are translated by Chrome in "div" instead of plain text */
+            if (reg.exec(html) !== null) {
+                var range = Selection.getSelectionRange(this.options.ownerDocument);
+                if (range !== null) {
+                    var parentNode = Selection.getSelectedParentElement(range),
+                        temp = html.replace(reg, '');
+                    /* Before the action is applied, alle the <br> are removed. */
+                    parentNode.innerHTML = temp;
+                    /* The action is applied so the markup does not get weird */
+                    this.options.ownerDocument.execCommand(action, false, null);
+                    /* Replace the html with the actual html with the <br> so the formatting
+                     * of the new line is not lost */
+                    return parentNode.innerHTML = html;
```

Comment:
Uhm, I don't think so because the problem is related to the wrapper tags; I try to explain myself: 
`h3 > text` become `h3 > div div`. This is about the wrappers of the text, not the text itself. I don't know if you got my point.